File: Cpp/434_Number_of_Segments_in_a_String.cpp
class Solution {
public:
    int countSegments(string s) {
        int result = 0;
        bool is_privious_space = true;
        for(auto& ch : s){
            if(!is_privious_space && ch == ' '){
                ++result;
            }
            is_privious_space = ch == ' ';
        }
        
        return result;
    }
};

File: Kotlin/434_Number_of_Segments_in_a_String.kt
class Solution {
    fun countSegments(s: String): Int {
        var result = 0
        var isPriviousSpace = true
        
        for(ch in s){
            if(isPriviousSpace && ch != ' '){
                ++result
            }
            
            isPriviousSpace = ch == ' '
        }
        
        return result
    }
}